cranelift/codegen/src/ir/pcc.rs

```diff
@@ -863,6 +1269,53 @@ impl<'a> FactContext<'a> {
         }
         Ok(())
     }
+
+    /// Apply a known inequality to rewrite dynamic bounds using transitivity, if possible.
+    ///
+    /// Given that `lhs >= rhs` (if not `strict`) or `lhs < rhs` (if
+    /// `strict`), update `fact.
+    pub fn apply_inequality(&self, fact: &Fact, lhs: &Fact, rhs: &Fact, strict: bool) -> Fact {
```

Comment:
Can we define an `enum` to replace the `strict: bool` parameter? This should both make this code easier to follow and make it more obvious what callers are doing vs just passing a random bool parameter or needing to do stuff like `foo(/* strict = */ false)`.

Something like

```rust
enum InequalityKind {
    GreaterThanOrEqual,
    LessThan,
}
```

Reply:
Also, it seems like maybe the comment is incorrect and this is `<=` or `<`, not `>=` or `<`?

Reply:
Done on both -- indeed, the comment was incorrect in the second case, thanks!